File: client/src/app/models/data-models/checklistTypePreferences.ts
export interface ChecklistTypePreferences {
    checklistTypePreferencesID: number;
    userID: number;
    key: string;
    label: string;
    isVisible?: boolean;
    isHeader: boolean;
    category: string;
}


export interface ChecklistTypePrefDto {
    checklistTypePreferencesID: number;
    isVisible: boolean;
}

File: API/Models/Tables/Entities/ChecklistTypePreferences.cs
namespace API.Models.Tables.Entities {
    public class ChecklistTypePreferences : BaseEntity {
        public int checklistTypePreferencesID { get; set; }
        public int userID { get; set; }
        public int checklistTypeID { get; set; }
        public bool? isVisible { get; set; }
        public bool? canUpdateQuestions { get; set; }
    }
}